src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -168,12 +173,22 @@ private CloudWatchCollector(JSONObject config, AmazonCloudWatchClient client) th
     }
 
     /**
-     * Check if a metric should be used according to `aws_dimension_select`
+     * Check if a metric should be used according to `aws_dimension_select` or `aws_dimension_select_regex`
      */
     private boolean useMetric(MetricRule rule, Metric metric) {
-      if (rule.awsDimensionSelect == null) {
+      if (rule.awsDimensionSelect == null && rule.awsDimensionSelectRegex == null) {
+        return true;
+      }
+      if (metricsIsInAwsDimensionSelect(rule, metric) || metricIsInAwsDimensionSelectRegex(rule, metric)) {
```

Comment:
This should be an and I'd say (not that it'd make much sense - maybe forbid that as a config error?)